litestar/handlers/base.py

```diff
@@ -345,76 +414,72 @@ def resolve_signature_namespace(self) -> dict[str, Any]:
             self._resolved_signature_namespace = ns
         return cast("dict[str, Any]", self._resolved_signature_namespace)
 
-    def resolve_dto(self) -> type[DTOInterface] | None:
+    def resolve_data_dto(self) -> type[AbstractDTO] | None:
         """Resolve the data_dto by starting from the route handler and moving up.
         If a handler is found it is returned, otherwise None is set.
         This method is memoized so the computation occurs only once.
 
         Returns:
-            An optional :class:`DTO type <.dto.interface.DTOInterface>`
+            An optional :class:`DTO type <.dto.base_dto.AbstractDTO>`
         """
-        if self._resolved_dto is Empty:
-            dtos: list[type[DTOInterface] | None] = [
-                layer_dto  # type:ignore[misc]
-                for layer in self.ownership_layers
-                if (layer_dto := layer.dto) is not Empty
-            ]
-            self._resolved_dto = dtos[-1] if dtos else None
+        if self._resolved_data_dto is Empty:
+            if data_dtos := cast(
+                "list[type[AbstractDTO] | None]",
+                [layer.dto for layer in self.ownership_layers if layer.dto is not Empty],
+            ):
+                data_dto: type[AbstractDTO] | None = data_dtos[-1]
+            elif self.parsed_data_field and (
+                plugins_for_data_type := [
+                    plugin
+                    for plugin in self.app.plugins.serialization
+                    if self.parsed_data_field.test_predicate_recursively(plugin.supports_type)
+                ]
+            ):
+                data_dto = plugins_for_data_type[0].create_dto_for_type(self.parsed_data_field)
+            else:
+                data_dto = None
+
+            if self.parsed_data_field and data_dto:
+                data_dto.create_for_field_definition(
+                    field_definition=self.parsed_data_field, handler_id=self.handler_id
+                )
+
```

Comment:
This should happen somewhere else within a separate method. This method is supposed to resolve DTOs from the ownership layers. Currently it does:

- Resolve existing DTOs from ownership layers
- Handle serialization plugins
- Create a new DTO if needed

An appropriate name would therefore be `resolve_and_create_data_dto_and_handle_serialization_plugins`, which is a bit lengthy in my opinion :upside_down_face:

Reply:
I disagree. It should resolve the DTOs for the handler. This logic means actually setting the providers for the handler, doing it once and caching.